### PR TITLE
Refactor GC

### PR DESCRIPTION
**Pull request by zhengxiexie**
Refactor GC

Add a common GenericGarbageCollector.
Replace sets.Set[string] with sets.String, since it supports Difference method, avoiding iterating in GC.

### CODE REVIEW

File: pkg/nsx/services/common/types.go [hunk 1]
```diff
@@ -167,6 +168,12 @@ type Service struct {
 	Client    client.Client
 	NSXClient *nsx.Client
 	NSXConfig *config.NSXOperatorConfig
+	Cancel    context.Context
```
Comment (timdengyun) [inline pkg/nsx/services/common/types.go:171]: where we use Cancel?

File: pkg/nsx/services/common/types.go [hunk 2]
```diff
@@ -167,6 +168,12 @@ type Service struct {
...
+}
+
+type ServiceAccount struct {
```
Comment (timdengyun) [inline pkg/nsx/services/common/types.go:174]: ditto

File: pkg/nsx/services/common/types.go (hunk 1 above)
Comment (zhengxiexie) [reply pkg/nsx/services/common/types.go:171]: oh, should removed.

**Review by TaoZou1 — DISMISSED** [review]

File: pkg/nsx/services/common/types.go (hunk 2 above)
Comment (timdengyun) [reply pkg/nsx/services/common/types.go:172]: do we need this struct?

Comment (zhengxiexie) [reply pkg/nsx/services/common/types.go]: fixed

File: pkg/nsx/services/common/store.go [hunk 3]
```diff
@@ -27,7 +27,7 @@ type Store interface {
 	// to specific nsx-t side resource and then add it to the store.
 	TransResourceToStore(obj *data.StructValue) error
 	// ListIndexFuncValues is the method to list all the values of the index
-	ListIndexFuncValues(key string) sets.Set[string]
+	ListIndexFuncValues(key string) sets.String //nolint:staticcheck // Ignore the deprecation warning for sets.String
```
Comment (timdengyun) [inline pkg/nsx/services/common/store.go:30]: I think the current sets.Set[string] is right way, instead of reverting to deprecated sets.String

**Pull request closed without merging**